contracts/settling_game/L02_Resources.cairo

```diff
@@ -252,6 +251,66 @@ func pillage_resources{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_c
     return ()
 end
 
+# @notice Mint resources from wonder
+# @param token_id: Staked realm id
+@external
+func wonder_claim{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    token_id : Uint256
+):
+    alloc_locals
+    let (caller) = get_caller_address()
+
+    # CONTRACT ADDRESSES
+    let (realms_address) = Module.get_external_contract_address(ExternalContractIds.Realms)
+    let (s_realms_address) = Module.get_external_contract_address(ExternalContractIds.S_Realms)
+    let (resources_address) = Module.get_external_contract_address(ExternalContractIds.Resources)
+    let (wonder_address) = Module.get_module_address(ModuleIds.L05_Wonders)
+
+    # FETCH OWNER
+    let (owner) = realms_IERC721.ownerOf(s_realms_address, token_id)
+
+    with_attr error_message("RESOURCES: ONLY S_REALM OWNER CAN CALL"):
+        assert caller = owner
+    end
+
+    # CALC DAYS
+    let (total_days, remainder) = days_accrued(token_id)
```

Comment:
can remove from this function from here and pass the `days` down into the `wonder_claim`